cmd/git_utils.go

```diff
@@ -99,31 +103,67 @@ func GetGitInfo(config *RootCommandConfig) (GitInfo, error) {
 
 	config.Debug.log("git version: ", version)
 
-	kargs := []string{"status", "-sb"}
-
-	output, gitErr := RunGit(config.LoggingConfig, config.ProjectDir, kargs, config.Dryrun)
+	gitInfo.Commit, gitErr = RunGitGetLastCommit(gitInfo.RemoteURL, config)
 	if gitErr != nil {
-		return gitInfo, gitErr
+		errMsg += "Received error getting current commit: " + gitErr.Error()
 	}
 
 	lineSeparator := "\n"
 	if runtime.GOOS == "windows" {
 		lineSeparator = "\r\n"
 	}
-	output = strings.Trim(output, trimChars)
-	outputLines := strings.Split(output, lineSeparator)
+
+	outputLines, err := RunGitBranchContains(config.LoggingConfig, gitInfo.Commit.SHA, config.ProjectDir, lineSeparator, config.Dryrun)
+	if err != nil {
+		errMsg += "Unable to locate latest commit in remote. "
+	} else {
+		gitInfo.Upstream = outputLines[0]
+		for _, upstream := range outputLines {
+			if strings.Contains(upstream, "origin") {
+				gitInfo.Upstream = upstream
+				break
+			} else if strings.Contains(upstream, "upstream") {
+				gitInfo.Upstream = upstream
+			}
+		}
+		gitInfo.Upstream = strings.TrimSpace(gitInfo.Upstream)
+	}
+
+	if gitInfo.Upstream == "" {
+		gitInfo.Commit.Pushed = false
+		gitRemoteOutput, err := RunGitRemote(config.LoggingConfig, config.ProjectDir, lineSeparator, config.Dryrun)
+		if err != nil {
+			return gitInfo, err
+		}
+
+		for _, remotes := range gitRemoteOutput {
+			if strings.Contains(remotes, "origin") {
```

Comment:
we might want to be more strict with this condition, maybe equals. Also would rename `remotes` to `remote`.